### ISSUE TITLE
Sourcemaps are off with Nollup in a Svelte project

### ISSUE THREAD

**Issue by rixo** [PepsRyuu/nollup#77]
Can be seen in my [svelte template](https://github.com/rixo/svelte-template-hot/tree/repro-nollup-sourcemaps). On the branch `repro-nollup-sourcemaps`, I have removed all my hot stuff from the Rollup config to reduce debugging noise.

I have put 2 `debugger` in a svelte file to clearly illustrate the problem.

Run `npm run dev`, and the debugger stops on the right mapped lines in the original svelte source.

Run `npm run dev:nollup` and everything works but the sourcemaps are off.

Both commands serves the app on http://localhost:5000.

For a little background, I have recently produced a Svelte HMR demo with Snowpack that gives instant feedback over HMR in a project that loads 500 Svelte components on the index page. This has revived interest a little for Svelte HMR in the community. I had the idea to subject Nollup to the same test. Just as fast. Twice faster on full reloads (since Snowpack sends 500+ files to the browser). I'm particularly happy to confirm that Nollup rebuild time is not affected by the size of the project, like the no-bundle dev servers (but unlike Rollup unfortunately). And so all this got me motivated to turn back to Nollup and push it as the preferred HMR solution for Svelte projects. This sourcemap issue has been the main bother for me.

**Comment by PepsRyuu** [comment]
Thanks for the reproduction, I can see what you mean.

It's likely something wrong with https://github.com/PepsRyuu/nollup/blob/master/lib/impl/utils.js#L61 It's always been a hack causing a variety of issues, and documentation on this seems to be very sparse. 

I'll try my best anyways and see what I can do! If anyone is reading this and has a good understanding of collapsing source maps, please let me know! :)

**Comment by PepsRyuu** [comment]
Figured it out, it's because of multi-line imports. 

![image](https://user-images.githubusercontent.com/6637882/81750157-e07c2080-94a4-11ea-9896-64f4dd4db3f7.png)

Commenting out the import rather than turning it to an empty string did the trick. Going to dig a bit deeper and see if there's a cleaner way of fixing this.

![image](https://user-images.githubusercontent.com/6637882/81750262-115c5580-94a5-11ea-8870-4cc6b6914e1b.png)

**Comment by rixo** [comment]
Wow :heart_eyes: This was fast. I must confess I had already sunk a handful of hours into this in the past and just found nothing... I was too focused on the hypothesis that the map collapse must have been buggy somehow :-x

I guess you could just turn it all to blank lines, but honestly commenting it out seems pretty clean to me. It cannot hurt to have it here when you have to dive into no sourcemap mode... Which happens now and then with Svelte, due to the very aggressive transformation of the original code.

**Comment by PepsRyuu** [comment]
If you want to double confirm it works, this is what I changed:

https://github.com/PepsRyuu/nollup/blob/master/lib/impl/ImportExportResolver.js#L178

I replaced it with this:

```
output.transpiled.appendLeft(node.start, '/*');
output.transpiled.appendRight(node.end, '*/');
```

**Comment by rixo** [comment]
Yes, it does!

It kills my Svelte HMR though... Apparently it misses an import (`Uncaught SyntaxError: Cannot use import statement outside a module`), but I don't see it myself in the eval'd code :thinking: Investigating...

Ah ah, I'm thrilled! Just earlier this afternoon, I told in the chat that Nollup was very well maintained, this 1h fix is gonna make my point!

**Comment by PepsRyuu** [comment]
```}
/*import * as ___SVELTE_HMR_HOT_API from 'svelte-template-hot/node_modules/rollup-plugin-svelte-hot/hmr/runtime.js';/**/import ___SVELTE_HMR_HOT_API_PROXY_ADAPTER from 'svelte-template-hot/node_modules/svelte-hmr/runtime/proxy-adapter-dom.js';*/
__e__('default', App
```
Comments are in the wrong position somehow...

**Comment by rixo** [comment]
Got it! It's this line:

~~~js
import * as ___SVELTE_HMR_HOT_API from '...';import ___SVELTE_HMR_HOT_API_PROXY_ADAPTER from '...';
~~~js

It gets transformed to:

~~~js
/*import * as ___SVELTE_HMR_HOT_API from \'...\';/**/import ___SVELTE_HMR_HOT_API_PROXY_ADAPTER from \'...\';*/\n\
~~~

Since the 2 import statement nodes are stuck together on the same line, the second import real `node.start` has been offset by the first one...

This is some code I own so I can easily fix it by adding some space, but it is better to fix it for everybody... Maybe just blanking them out would be better in the end? I don't see an easy way to patch the `node.start`, if magic-string doesn't already do it...

(Beaten to the post ^^)

**Comment by PepsRyuu** [comment]
That's the thing, I'm wondering why ```magic-string``` isn't working correctly here. Will have to check the library and see if this is a known issue.

**Comment by PepsRyuu** [comment]
```
output.transpiled.overwrite(node.start, node.end, input.substring(node.start, node.end).replace(/[^\n\r]/g, ' '));
    return importee;
```

Might be a faster approach that I'm not thinking of.

**Comment by rixo** [comment]
Yup. Works for me too.

**Comment by PepsRyuu** [comment]
Just looking into this more, this is going to require a bit of a refactor for ImportExportResolver, so might be a few days before it's done fully. :)

**Comment by PepsRyuu** [comment]
I've created a PR that uses the blanking approach: https://github.com/PepsRyuu/nollup/pull/81

Feel free to try it out! Will merge in a couple of days once I'm confident it won't break anything. :)

**Comment by PepsRyuu** [comment]
Merged, release to follow after other small features merged.

**Comment by PepsRyuu** [comment]
Released in ```0.12.0```

**Comment by rixo** [comment]
> Merged, release to follow after other small features merged.

> Released in 0.12.0

Congrats for 0.12.0 and all those "small features"! :clinking_glasses: :tada: Thanks for you hard work!
